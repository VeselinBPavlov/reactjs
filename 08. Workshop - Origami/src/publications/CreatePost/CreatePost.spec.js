import React from "react";
import CreatePost from "./CreatePost";
import { shallow } from "enzyme";
import postService from '../../services/post-service';

it("renders create post component", () => {
  const wrapper = shallow(<CreatePost />);
  expect(wrapper).toBeTruthy();
});

it("checks for html changes", () => {
  const wrapper = shallow(<CreatePost />);
  expect(wrapper).toMatchSnapshot();
});

it("creates post successfully", (done) => {
  const history = { push: () => {} };
  const useRefSpy = jest.spyOn(React, 'useRef').mockReturnValueOnce({ current: { value: 123 } });
  const historyPushSpy = jest.spyOn(history, 'push');
  const createPostSpy = jest.spyOn(postService, 'create').mockImplementationOnce(() => Promise.resolve());;

  const wrapper = shallow(<CreatePost history={history}/>);
  wrapper.find('button').simulate('click');

  expect(useRefSpy).toHaveBeenCalledTimes(1);
  expect(createPostSpy).toHaveBeenCalledTimes(1);
  expect(createPostSpy).toHaveBeenCalledWith({ description: 123 });
  Promise.resolve().then(() => {
    expect(historyPushSpy).toHaveBeenCalledTimes(1);
    done();
    useRefSpy.mockClear();
  });
});