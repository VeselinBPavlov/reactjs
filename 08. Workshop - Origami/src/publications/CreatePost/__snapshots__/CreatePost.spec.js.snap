// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks for html changes 1`] = `
<div
  className="CreatePost"
>
  <form>
    <textarea />
    <button
      onClick={[Function]}
      type="button"
    >
      Create Post
    </button>
  </form>
  <Posts
    limit={3}
  />
</div>
`;
